extension.js

```diff
@@ -912,17 +913,48 @@ function handle_doc_open(active_doc) {
 
 function quote_field(field, delim) {
     if (field.indexOf('"') != -1 || field.indexOf(delim) != -1) {
-        return '"' + field.replace(/"/g, '""')  + '"';
+        return '"' + field.replace(/"/g, '""') + '"';
     }
     return field;
 }
 
 
 function quoted_join(fields, delim) {
-    var quoted_fields = fields.map(function(val) { return quote_field(val, delim); });
+    var quoted_fields = fields.map(function (val) { return quote_field(val, delim); });
     return quoted_fields.join(delim);
 }
 
+function sample(uri) {
+    var filePath = uri.fsPath;
+
+    var sizeLimit = 102400; // 100 KB
+    var fileSizeInBytes = fs.statSync(filePath)['size'];
+    if (fileSizeInBytes < sizeLimit) {
+        vscode.window.showInformationMessage('No need to cut.')
+        return;
+    }
+
+    var rl = readline.createInterface({
+        input: fs.createReadStream(filePath, {
+            start: 0,
+            end: sizeLimit
+        }),
+        crlfDelay: Infinity
+    });
+
+    var lines = [];
+
+    rl.on('line', (line) => {
+        lines.push(line);
+    });
+
+    rl.on('close', () => {
+        lines.pop();
+        const outPath = filePath.replace('.csv', '.100kb.csv').replace('.tsv', '.100kb.tsv');
```

Comment:
For now, we only register this "sample" command when `resourceLangId == csv/tsv`. And I just checked that even if we change the editor language to `csv` from the right bottom of vscode, `txt` file doesn't get this command in the context menu.

(Of course it is a lazy implementation since for me `.csv` is enough 😂)

Reply:
let's simplify the naming scheme: `const outPath = '.rb_csv_preview.' + filePath`. Also please use the same coding style as I do, i.e. `out_path` instead of `outPath`, to be consistent with the rest of the code.